### PR TITLE
python3Packages.pycm: init at 2.5

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [NA] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/pycm/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner  = "sepandhaghighi";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "0zfv20hd7zq95sflsivjk47b0sm7q76w7fv2i2mafn83ficzx0p0";
+  };
+
+  # remove a trivial dependency on the author's `art` Python ASCII art library
+  patchPhase = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/pycm/default.nix]: postPatch

File: pkgs/development/python-modules/pycm/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    rm pycm/__main__.py
+    substituteInPlace setup.py --replace '=get_requires()' '=[]'
+  '';
+
+  checkInputs = [ pytest ];
+  buildInputs = [ coverage ];
```
Comment (FRidh) [inline pkgs/development/python-modules/pycm/default.nix]: It's more likely this is for `checkInputs`

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pycm/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pycm/default.nix]: Doesn't seem needed actually...

File: pkgs/development/python-modules/pycm/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pycm/default.nix]: Changed.

**Review by doronbehar — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
diff LGTM
has tests :+1: 

```
[4 built, 22 copied (50.0 MiB), 9.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/77275
2 package were built:
python37Packages.pycm python38Packages.pycm
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.pycm python38Packages.pycm

**Pull request merged**